src/popup/SegmentListComponent.tsx

```diff
@@ -46,15 +54,25 @@ export const SegmentListComponent = (props: SegmentListComponentProps) => {
         }
     }, []);
 
+    const [hasSegments, hasChapters] = React.useMemo(() => {
+        const hasSegments = Boolean(props.segments.find(isSegment))
+        const hasChapters = Boolean(props.segments.find(isChapter))
+        return [hasSegments, hasChapters];
+    }, [props.segments]);
+
     React.useEffect(() => {
-        setTab(SegmentListTab.Segments);
-    }, [props.videoID]);
+        if (hasSegments){
+            setTab(SegmentListTab.Segments);
+        } else {
+            setTab(SegmentListTab.Chapter);
+        }
+    }, [props.videoID, hasSegments]);
 
     const tabFilter = (segment: SponsorTime) => {
         if (tab === SegmentListTab.Chapter) {
-            return segment.actionType === ActionType.Chapter;
+            return isChapter(segment);
         } else {
-            return segment.actionType !== ActionType.Chapter;
+            return isSegment(segment);
         }
     };
```

Comment:
since you made these two branches into their own functions, you don't really need the wrapping function anymore - you can just assign the right function to the `tabFilter` variable directly